lib/ansible/modules/remote_management/dellemc/idrac/dellemc_idrac_firmware.py

```diff
@@ -0,0 +1,216 @@
+       job_wait:   True
+       catalog_file_name:  "Catalog.xml"
+"""
+
+RETURN = """
+---
+msg:
+    description: Updates firmware from a repository on a network share (CIFS, NFS).
+    returned: success
+    type: dict
+    sample: {
+        'CompletionTime': '2018-01-24T00:47:33',
+        'Id': 'JID_XXXXXXXXXXXX',
+        'JobState': 'Completed',
+        'JobType': 'ImportConfiguration',
+        'Message': 'Successfully imported and applied Server Configuration Profile.',
+        'MessageId': 'SYSXXX',
+        'Name': 'Import Configuration',
+        'StartTime': 'TIME_NOW',
+        'Status': 'Success',
+    }
+"""
+
+
+from ansible.module_utils.remote_management.dellemc.dellemc_idrac import iDRACConnection
+from ansible.module_utils.basic import AnsibleModule
+try:
+    from omsdk.sdkcreds import UserCredentials
+    from omsdk.sdkfile import FileOnShare
+    HAS_OMSDK = True
+except ImportError:
+    HAS_OMSDK = False
+
+
+def _validate_catalog_file(catalog_file_name):
+    normilized_file_name = catalog_file_name.lower()
+    if not normilized_file_name:
+        raise ValueError('catalog_file_name should be a non-empty string.')
+    elif not normilized_file_name.endswith("xml"):
+        raise ValueError('catalog_file_name should be an XML file.')
+
+
+def update_firmware(idrac, module):
+    """Update firmware from a network share and return the job details."""
+    msg = {}
+    msg['changed'] = False
+    msg['msg'] = {}
+    err = False
+
+    try:
+        _validate_catalog_file(module.params['catalog_file_name'])
+    except ValueError as e:
+        module.fail_json(msg=e.message)
+
+    try:
+        upd_share = FileOnShare(remote=module.params['share_name'] + "/" + module.params['catalog_file_name'],
+                                mount_point=module.params['share_mnt'],
+                                isFolder=False,
+                                creds=UserCredentials(
+                                    module.params['share_user'],
+                                    module.params['share_pwd'])
+                                )
+
+        idrac.use_redfish = True
+        if '12' in idrac.ServerGeneration or '13' in idrac.ServerGeneration:
+            idrac.use_redfish = False
+
+        apply_update = True
+        msg['msg'] = idrac.update_mgr.update_from_repo(upd_share,
+                                                       apply_update,
+                                                       module.params['reboot'],
+                                                       module.params['job_wait'])
+
+        if "Status" in msg['msg']:
+            if msg['msg']['Status'] == "Success":
+                if module.params['job_wait'] is True:
+                    msg['changed'] = True
+            else:
+                err = True
+
+    except Exception as e:
+        err = True
+        msg['msg'] = str(e)
+
+    return msg, err
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
```

Comment:
Could you please replace `dict()` constructors with literals (`{}`)

Reply:
done